presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java

```diff
@@ -52,6 +53,13 @@ public class HiveColumnHandle
     public static final HiveType BUCKET_HIVE_TYPE = HIVE_INT;
     public static final TypeSignature BUCKET_TYPE_SIGNATURE = BUCKET_HIVE_TYPE.getTypeSignature();
 
+    public static final int ROW_ID_COLUMN_INDEX = -10;
+    public static final String ROW_ID_COLUMN_NAME = "$row_id";
+    public static final HiveType ROW_ID_TYPE = HIVE_BINARY;
+    public static final TypeSignature ROW_ID_TYPE_SIGNATURE = ROW_ID_TYPE.getTypeSignature();
+
+    // TODO these are buggy. Column indexes less than -13 can conflict with regular column indexes.
+    // Renumber them if possible.
```

Comment:
I expect these should be trivial to renumber (it's entirely an internal concept, and should just involve changing the number).  You could add a commit here that does it or as a follow up.